From: NIIBE Yutaka <gniibe@fsij.org>
Date: Thu, 21 Feb 2019 12:26:09 +0900
Subject: tests: Add "disable-scdaemon" in gpg-agent.conf.

* tests/openpgp/defs.scm: Add "disable-scdaemon".  Remove
  "scdaemon-program".
* tests/gpgme/gpgme-defs.scm, tests/gpgsm/gpgsm-defs.scm: Likewise.
* tests/inittests, tests/pkits/inittests: Add "disable-scdaemon"

--

Before this change, running "make check" accesses USB device by
scdaemon on host computer.  If there is any smartcard/token available,
it may affect test results.  Because default key choice depends on
smartcard/token availability now and existing tests have nothing about
testing smartcard/token, disabling scdaemon is good.

Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
(cherry picked from commit 64b7c6fd1945bc206cf56979633dfca8a7494374)
(cherry picked from commit 150d5452318eafa6aa800ff3b87f8f8eb35ed203)
---
 tests/gpgme/gpgme-defs.scm | 3 +--
 tests/gpgsm/gpgsm-defs.scm | 5 +----
 tests/inittests            | 1 +
 tests/openpgp/defs.scm     | 5 +----
 tests/pkits/inittests      | 1 +
 5 files changed, 5 insertions(+), 10 deletions(-)

diff --git a/tests/gpgme/gpgme-defs.scm b/tests/gpgme/gpgme-defs.scm
index 0de589f..bc40b3c 100644
--- a/tests/gpgme/gpgme-defs.scm
+++ b/tests/gpgme/gpgme-defs.scm
@@ -67,8 +67,7 @@
   (create-file
    "gpg-agent.conf"
    (string-append "pinentry-program " (tool 'pinentry))
-   (string-append "scdaemon-program " (tool 'scdaemon))
-   )
+   "disable-scdaemon")
 
   (start-agent)
 
diff --git a/tests/gpgsm/gpgsm-defs.scm b/tests/gpgsm/gpgsm-defs.scm
index f118642..848bc75 100644
--- a/tests/gpgsm/gpgsm-defs.scm
+++ b/tests/gpgsm/gpgsm-defs.scm
@@ -67,10 +67,7 @@
 	       "faked-system-time 1008241200")
   (create-file "gpg-agent.conf"
 	       (string-append "pinentry-program " (tool 'pinentry))
-	       (if (assoc "scdaemon" gpg-components)
-		   (string-append "scdaemon-program " (tool 'scdaemon))
-		   "# No scdaemon available")
-	       )
+	       "disable-scdaemon")
   (start-agent)
   (create-file
    "trustlist.txt"
diff --git a/tests/inittests b/tests/inittests
index 6fbccfb..9090674 100755
--- a/tests/inittests
+++ b/tests/inittests
@@ -85,6 +85,7 @@ EOF
 cat > gpg-agent.conf <<EOF
 no-grab
 pinentry-program /home/wk/work/pinentry/gtk/pinentry-gtk
+disable-scdaemon
 EOF
 
 cat > trustlist.txt <<EOF
diff --git a/tests/openpgp/defs.scm b/tests/openpgp/defs.scm
index 7e41d19..94c5beb 100644
--- a/tests/openpgp/defs.scm
+++ b/tests/openpgp/defs.scm
@@ -355,10 +355,7 @@
 	       (if (flag "--extended-key-format" *args*)
 		   "enable-extended-key-format" "#enable-extended-key-format")
 	       (string-append "pinentry-program " (tool 'pinentry))
-	       (if (assoc "scdaemon" gpg-components)
-		   (string-append "scdaemon-program " (tool 'scdaemon))
-		   "# No scdaemon available")
-	       ))
+	       "disable-scdaemon"))
 
 ;; Initialize the test environment, install appropriate configuration
 ;; and start the agent, without any keys.
diff --git a/tests/pkits/inittests b/tests/pkits/inittests
index 4bff0a8..deb1854 100755
--- a/tests/pkits/inittests
+++ b/tests/pkits/inittests
@@ -94,6 +94,7 @@ EOF
 # Fixme: we need to write a dummy pinentry program
 cat > gpg-agent.conf <<EOF
 no-grab
+disable-scdaemon
 EOF
 
 # Mark the root CA trusted
